core/src/avm2/domain.rs

```diff
@@ -58,22 +68,49 @@ impl<'gc> Domain<'gc> {
     /// You must initialize domain memory later on after the ByteArray class is
     /// instantiated but before user code runs.
     pub fn uninitialized_domain(mc: &Mutation<'gc>, parent: Option<Domain<'gc>>) -> Domain<'gc> {
-        Self(GcCell::new(
+        let domain = Self(GcCell::new(
             mc,
             DomainData {
                 defs: PropertyMap::new(),
                 classes: PropertyMap::new(),
                 parent,
                 domain_memory: None,
                 default_domain_memory: None,
+                children: Vec::new(),
             },
-        ))
+        ));
+        if let Some(parent) = parent {
+            parent
+                .0
+                .write(mc)
+                .children
+                .push(DomainWeak(GcCell::downgrade(domain.0)));
+        }
+        domain
+    }
+
+    pub fn classes(&self) -> Ref<'_, PropertyMap<'gc, GcCell<'gc, Class<'gc>>>> {
+        Ref::map(self.0.read(), |r| &r.classes)
     }
 
     pub fn is_playerglobals_domain(&self, avm2: &Avm2<'gc>) -> bool {
         avm2.playerglobals_domain.0.as_ptr() == self.0.as_ptr()
     }
 
+    pub fn children(&self, mc: &Mutation<'gc>) -> Vec<Domain<'gc>> {
+        // Take this opportunity to clean up dead children.
```

Comment:
So if the user never opens this new dialog, the children list will grow and grow with more and more dead weak refs?
Not that I assume domains come and go that often in large numbers, or that these refs would consume that much memory...
But am I getting this right?

Reply:
That's right - a domain should be created at most once per loaded swf (unless a game does something really weird), so I think this should be fine. We could skip adding it when the debug ui is disabled, but I think this could be useful for debugging